packages/components/src/CreditCard/Menu/Item.purs

```diff
@@ -1,55 +1,61 @@
 module KSF.CreditCard.Menu.Item where
 
-import Prelude (Unit, ($), (<>), show)
+import Prelude (Unit, ($), (<>), pure, show)
 
 import Bottega.Models (CreditCard, CreditCardId(..))
 import Data.String.CodePoints (splitAt)
 import Effect (Effect)
 import React.Basic (JSX)
-import React.Basic.Classic (make)
-import React.Basic.Classic as React
 import React.Basic.DOM as DOM
 import React.Basic.Events (handler_)
+import React.Basic.Hooks (Component)
+import React.Basic.Hooks as React
 
-type Self = React.Self Props State
+-- type Self = React.Self Props State
 
-type State = { selected :: Boolean }
+-- type State = { selected :: Boolean }
 
 type Props =
   { creditCard :: CreditCard
   , onClick :: Effect Unit
   }
 
-item :: Props -> JSX
-item = make component { initialState, render }
+-- item :: Props -> JSX
+-- item = make component { initialState, render }
 
-component :: React.Component Props
-component = React.createComponent "item"
+-- component :: React.Component Props
+-- component = React.createComponent "item"
 
-initialState :: State
-initialState = { selected: false }
+component :: Component Props
+component = React.component "item" $ \props -> React.do
+  pure $ render props
 
-render :: Self -> JSX
-render { props: { creditCard, onClick } } = DOM.label
-                { className: "credit-card-menu-item"
-                , children: [ DOM.input
-                                { type: "radio"
-                                , name: "credit-card-menu-item--selection"
-                                , onClick: handler_ $ onClick
-                                , value: creditCardId creditCard.id
-                                }
-                            , DOM.div
-                                { className: "credit-card-menu-item--radio-button"
-                                , id: creditCardId creditCard.id
-                                , children:
-                                    [ DOM.div { className: "credit-card-menu-item--radio-button_checked" } ]
-                                }
-                            , DOM.div_
-                                [ number creditCard.maskedPan
-                                , expiryDate creditCard.expiryDate
-                                ]
-                            ]
-                }
+-- initialState :: State
+-- initialState = { selected: false }
+
+render :: Props -> JSX
+render { creditCard, onClick } =
+  DOM.label
+    { className: "credit-card-menu-item"
+    , children:
+        [ DOM.input
+            { type: "radio"
+            , name: "credit-card-menu-item--selection"
+            , onClick: handler_ $ onClick
```

Comment:
Unneeded function application.

Reply:
If I remove `handler`, I get a type error:
```
Error found:
in module KSF.CreditCard.Menu.Item
at ./../../packages/components/src/CreditCard/Menu/Item.purs:28:11 - 33:14 (line 28, column 11 - line 33, column 14)

  Could not match type
          
    Effect
          
  with type
                            
    EffectFn1 SyntheticEvent
                            

while trying to match type Effect Unit
  with type EffectFn1 SyntheticEvent Unit
while matching label onClick
while solving type class constraint
                                                                        
  Prim.Row.Union ( name :: String                                       
                 , onClick :: Effect Unit                               
                 , type :: String                                       
                 , value :: String                                      
                 )                                                      
                 t0                                                     
                 ( _aria :: Object String                               
                 , _data :: Object String                               
                 , about :: String                                      
                 , accept :: String                                     
                 , acceptCharset :: String                              
                 , accessKey :: String                                  
                 , allowFullScreen :: Boolean                           
                 , allowTransparency :: Boolean                         
                 , alt :: String                                        
                 , autoCapitalize :: String                             
                 , autoComplete :: String                               
                 , autoCorrect :: String                                
                 , autoFocus :: Boolean                                 
                 , autoPlay :: Boolean                                  
                 , autoSave :: String                                   
                 , capture :: Boolean                                   
                 , cellPadding :: String                                
                 , cellSpacing :: String                                
                 , charSet :: String                                    
                 , checked :: Boolean                                   
                 , classID :: String                                    
                 , className :: String                                  
                 , colSpan :: Int                                       
                 , contentEditable :: Boolean                           
                 , contextMenu :: String                                
                 , crossOrigin :: String                                
                 , dangerouslySetInnerHTML :: { __html :: String        
                                              }                         
                 , datatype :: String                                   
                 , dateTime :: String                                   
                 , defaultChecked :: String                             
                 , defaultValue :: String                               
                 , dir :: String                                        
                 , disabled :: Boolean                                  
                 , draggable :: Boolean                                 
                 , encType :: String                                    
                 , form :: String                                       
                 , formAction :: String                                 
                 , formEncType :: String                                
                 , formMethod :: String                                 
                 , formNoValidate :: Boolean                            
                 , formTarget :: String                                 
                 , frameBorder :: String                                
                 , height :: String                                     
                 , hidden :: Boolean                                    
                 , hrefLang :: String                                   
                 , htmlFor :: String                                    
                 , httpEquiv :: String                                  
                 , icon :: String                                       
                 , id :: String                                         
                 , inlist :: String                                     
                 , inputMode :: String                                  
                 , is :: String                                         
                 , itemID :: String                                     
                 , itemProp :: String                                   
                 , itemRef :: String                                    
                 , itemScope :: Boolean                                 
                 , itemType :: String                                   
                 , key :: String                                        
                 , keyParams :: String                                  
                 , keyType :: String                                    
                 , lang :: String                                       
                 , list :: String                                       
                 , marginHeight :: String                               
                 , marginWidth :: String                                
                 , max :: String                                        
                 , maxLength :: Int                                     
                 , mediaGroup :: String                                 
                 , min :: String                                        
                 , minLength :: Int                                     
                 , multiple :: Boolean                                  
                 , name :: String                                       
                 , noValidate :: Boolean                                
                 , onAnimationEnd :: EffectFn1 SyntheticEvent Unit      
                 , onAnimationIteration :: EffectFn1 SyntheticEvent Unit
                 , onAnimationStart :: EffectFn1 SyntheticEvent Unit    
                 , onBlur :: EffectFn1 SyntheticEvent Unit              
                 , onChange :: EffectFn1 SyntheticEvent Unit            
                 , onClick :: EffectFn1 SyntheticEvent Unit             
                 , onCompositionEnd :: EffectFn1 SyntheticEvent Unit    
                 , onCompositionStart :: EffectFn1 SyntheticEvent Unit  
                 , onCompositionUpdate :: EffectFn1 SyntheticEvent Unit 
                 , onContextMenu :: EffectFn1 SyntheticEvent Unit       
                 , onCopy :: EffectFn1 SyntheticEvent Unit              
                 , onCut :: EffectFn1 SyntheticEvent Unit               
                 , onDoubleClick :: EffectFn1 SyntheticEvent Unit       
                 , onDrag :: EffectFn1 SyntheticEvent Unit              
                 , onDragEnd :: EffectFn1 SyntheticEvent Unit           
                 , onDragEnter :: EffectFn1 SyntheticEvent Unit         
                 , onDragExit :: EffectFn1 SyntheticEvent Unit          
                 , onDragLeave :: EffectFn1 SyntheticEvent Unit         
                 , onDragOver :: EffectFn1 SyntheticEvent Unit          
                 , onDragStart :: EffectFn1 SyntheticEvent Unit         
                 , onDrop :: EffectFn1 SyntheticEvent Unit              
                 , onFocus :: EffectFn1 SyntheticEvent Unit             
                 , onGotPointerCapture :: EffectFn1 SyntheticEvent Unit 
                 , onInvalid :: EffectFn1 SyntheticEvent Unit           
                 , onKeyDown :: EffectFn1 SyntheticEvent Unit           
                 , onKeyPress :: EffectFn1 SyntheticEvent Unit          
                 , onKeyUp :: EffectFn1 SyntheticEvent Unit             
                 , onLostPointerCapture :: EffectFn1 SyntheticEvent Unit
                 , onMouseDown :: EffectFn1 SyntheticEvent Unit         
                 , onMouseEnter :: EffectFn1 SyntheticEvent Unit        
                 , onMouseLeave :: EffectFn1 SyntheticEvent Unit        
                 , onMouseMove :: EffectFn1 SyntheticEvent Unit         
                 , onMouseOut :: EffectFn1 SyntheticEvent Unit          
                 , onMouseOver :: EffectFn1 SyntheticEvent Unit         
                 , onMouseUp :: EffectFn1 SyntheticEvent Unit           
                 , onPaste :: EffectFn1 SyntheticEvent Unit             
                 , onPointerCancel :: EffectFn1 SyntheticEvent Unit     
                 , onPointerDown :: EffectFn1 SyntheticEvent Unit       
                 , onPointerEnter :: EffectFn1 SyntheticEvent Unit      
                 , onPointerLeave :: EffectFn1 SyntheticEvent Unit      
                 , onPointerMove :: EffectFn1 SyntheticEvent Unit       
                 , onPointerOut :: EffectFn1 SyntheticEvent Unit        
                 , onPointerOver :: EffectFn1 SyntheticEvent Unit       
                 , onPointerUp :: EffectFn1 SyntheticEvent Unit         
                 , onSelect :: EffectFn1 SyntheticEvent Unit            
                 , onSubmit :: EffectFn1 SyntheticEvent Unit            
                 , onTouchCancel :: EffectFn1 SyntheticEvent Unit       
                 , onTouchEnd :: EffectFn1 SyntheticEvent Unit          
                 , onTouchMove :: EffectFn1 SyntheticEvent Unit         
                 , onTouchStart :: EffectFn1 SyntheticEvent Unit        
                 , onTransitionEnd :: EffectFn1 SyntheticEvent Unit     
                 , onWheel :: EffectFn1 SyntheticEvent Unit             
                 , pattern :: String                                    
                 , placeholder :: String                                
                 , prefix :: String                                     
                 , property :: String                                   
                 , radioGroup :: String                                 
                 , readOnly :: Boolean                                  
                 , ref :: Ref (Nullable Node)                           
                 , required :: Boolean                                  
                 , resource :: String                                   
                 , results :: String                                    
                 , role :: String                                       
                 , rowSpan :: Int                                       
                 , scoped :: Boolean                                    
                 , seamless :: Boolean                                  
                 , security :: String                                   
                 , size :: Int                                          
                 , slot :: String                                       
                 , spellCheck :: Boolean                                
                 , src :: String                                        
                 , srcDoc :: JSX                                        
                 , srcLang :: String                                    
                 , srcSet :: String                                     
                 , step :: String                                       
                 , style :: CSS                                         
                 , suppressContentEditableWarning :: Boolean            
                 , tabIndex :: Int                                      
                 , title :: String                                      
                 , title :: String                                      
                 , type :: String                                       
                 , typeof :: String                                     
                 , unselectable :: Boolean                              
                 , useMap :: String                                     
                 , value :: String                                      
                 , vocab :: String                                      
                 , width :: String                                      
                 , wmode :: String                                      
                 )                                                      
                                                                        
while inferring the type of input { type: "radio"                           
                                  , name: "credit-card-menu-item--selection"
                                  , onClick: onClick                        
                                  , value: creditCardId (...)               
                                  }                                         
in value declaration render

where t0 is an unknown type
```

Reply:
The only thing that can be removed is the `$`. Maybe that's what you meant?